src/hipscat_import/catalog/resume_plan.py

```diff
@@ -0,0 +1,224 @@
+    """set of files (and job keys) that have yet to be split"""
+
+    MAPPING_START_LOG_FILE = "mapping_start_log.txt"
+    MAPPING_DONE_LOG_FILE = "mapping_done_log.txt"
+    SPLITTING_LOG_FILE = "splitting_log.txt"
+    HISTOGRAM_BINARY_FILE = "mapping_histogram.binary"
+    REDUCING_LOG_FILE = "reducing_log.txt"
+
+    MAPPING_DONE_FILE = "mapping_done"
+    SPLITTING_DONE_FILE = "splitting_done"
+    REDUCING_DONE_FILE = "reducing_done"
+
+    def __post_init__(self):
+        """Initialize the plan."""
+        self.gather_plan()
+
+    def gather_plan(self):
+        """Initialize the plan."""
+        with tqdm(
+            total=4, desc="Planning ", disable=not self.progress_bar
+        ) as step_progress:
+            ## Make sure it's safe to use existing resume state.
+            if not self.resume:
+                if file_io.directory_has_contents(self.tmp_path):
+                    raise ValueError(
+                        f"tmp_path ({self.tmp_path}) contains intermediate files."
+                        " choose a different directory or use --resume flag"
+                    )
+            file_io.make_directory(self.tmp_path, exist_ok=True)
+            step_progress.update(1)
+
+            ## Validate existing resume state.
+            ## - if a later stage is complete, the earlier stages should be complete too.
+            mapping_done = self.is_mapping_done()
+            splitting_done = self.is_splitting_done()
+            reducing_done = self.is_reducing_done()
+
+            if reducing_done and (not mapping_done or not splitting_done):
+                raise ValueError(
+                    "mapping and splitting must be complete before reducing"
+                )
+            if splitting_done and not mapping_done:
+                raise ValueError("mapping must be complete before splitting")
+            step_progress.update(1)
+
+            ## Gather keys for execution.
+            self.input_paths.sort()
+            for test_path in self.input_paths:
+                if not file_io.does_file_or_directory_exist(test_path):
+                    raise FileNotFoundError(f"{test_path} not found on local storage")
+            step_progress.update(1)
+            if not mapping_done:
+                mapping_start_keys = self._read_log_keys(self.MAPPING_START_LOG_FILE)
+                mapping_done_keys = self._read_log_keys(self.MAPPING_DONE_LOG_FILE)
+                if len(mapping_done_keys) != len(mapping_start_keys):
+                    raise ValueError(
+                        "Resume logs are corrupted. "
+                        "Delete temp directory and restart import pipeline."
+                    )
+                mapped_paths = set(mapping_start_keys)
+                self.map_files = [
+                    file_path
+                    for file_path in self.input_paths
+                    if f"map_{file_path}" not in mapped_paths
```

Comment:
Can generating the key like this be done in a separate function? Just to make sure it's consistent. (Not that it isn't right now, but in case we want to use it in the future)